vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/LinkCrawler.java

```diff
@@ -33,14 +37,16 @@ public class LinkCrawler extends WebCrawler
 {
     private static final Logger LINK_CRAWLER_LOGGER = LoggerFactory.getLogger(LinkCrawler.class);
 
-    private static final String[] EXCLUDED_EXTENSIONS = { ".css", ".gif", ".gz", ".ico", ".jpeg", ".jpg", ".js",
-            ".mp3", ".mp4", ".pdf", ".png", ".zip" };
-
     private final LinkCrawlerData linkCrawlerData;
 
-    public LinkCrawler(LinkCrawlerData linkCrawlerData)
+    private final Optional<Pattern> excludeExtensionsPattern;
+
+    private final Set<String> urlsWithInvalidRedirects = new HashSet<>();
```

Comment:
in general is it possible to catch race condition here?

Reply:
> in general is it possible to catch race condition here?

Each instance of `LinkCrawler` runs in a single thread, so inside this class race condition is not possible